File: Builder Pattern in JavaScript/QueryBuilder.js
const Query = require('./Query');

class QueryBulder {

    #query;

    constructor() {
        this.#query = new Query();
    }

    select(select) {
        this.#query.select = select;
        return this;
    }

    from(from) {
        this.#query.from = from;
        return this;
    }

    where(where) {
        this.#query.where = where;
        return this;
    }

    build() {
        return this.#query;
    }

}

module.exports = QueryBulder;

File: Design Patterns/Builder Pattern/QueryBuilder.java
package querybuilder;

import query.Query;

public class QueryBuilder{
	
	private Query query;
	
	public QueryBuilder(){
		query = new Query();
	}
	
	public QueryBuilder select(String select){
		query.setSelect(select);
		return this;
	}
	
	public QueryBuilder from(String from){
		query.setFrom(from);
		return this;
	}	
	
	public QueryBuilder where(String where){
		query.setWhere(where);
		return this;
	}
	
	public Query build(){
		return query;
	}
	
	
}